tools/shared.py

```diff
@@ -1240,7 +1241,8 @@ def is_exe(fpath):
     if fpath:
       if is_exe(program): return program
     else:
-      for path in os.environ["PATH"].split(os.pathsep):
+      paths = os.environ["PATH"].split(os.pathsep) + (extra_paths or [])
```

Comment:
hmm, why not do `extra_paths=[]` in the declaration? then can just write `extra_paths` on this line?